test/coverlet.core.tests/CoverageTests.cs

```diff
@@ -106,5 +106,35 @@ public void Condition_If()
                 File.Delete(path);
             }
         }
+
+        [Fact]
+        public void SelectionStatements_Switch()
+        {
+            string path = Path.GetTempFileName();
+            try
+            {
+                RemoteExecutor.Invoke(async pathSerialize =>
+                {
+                    CoveragePrepareResult coveragePrepareResult = await TestInstrumentationHelper.Run<SelectionStatements>(instance =>
+                    {
+                        instance.Switch(1);
+                        return Task.CompletedTask;
+                    }, pathSerialize);
+                    return 0;
+                }, path).Dispose();
+
+                CoverageResult result = TestInstrumentationHelper.GetCoverageResult(path);
+
+                result.Document("Instrumentation.SelectionStatements.cs")
+                      .AssertLinesCovered(BuildConfiguration.Release, (24, 1), (26, 0), (28, 0))
+                      .AssertBranchesCovered(BuildConfiguration.Release, (24, 1, 1))
```

Comment:
This looks like it make the test very dependent on the optimisations in the compiler, so they could start breaking on a different SDK version. Since these tests are about matching specific lines it seems better to just use Debug runs and not include any expected outcome for Release runs.

Reply:
> This looks like it make the test very dependent on the optimisations in the compiler,

Yep it's true...talked also with Andrew https://github.com/tonerdo/coverlet/pull/396#issuecomment-495944424 and we had same issue in past https://github.com/tonerdo/coverlet/issues/389
Yes we could test only for debug config...but why if some bug happen on release and not on debug config for some reason related to "instrumentation" bug?If in future we put some conditional code related to some new roslyn behaviour?Maybe I'm too concern about something that won't never happen 😅 , but could be a good "doorbell", and maybe we could add the possibility to run test only for specific SDK like a custom xunit attribute `[RunOnSdks("2.2.301","..")]`, to be very rigorous.
Another thing that I'm not sure is if also "debug" config will be stable...I mean also that config could depend on `different SDK version`, less likely but I'm not 100% sure.
**BTW I don't have a super strong opinion and we could test only debug config**...I tried to be complete as possible...
Thank's for review Peter!!!
@tonerdo thoughts?